### DISCUSSION TITLE
make fails with ld errors

### DISCUSSION THREAD

**Discussion by qw3r3wq** [dawigit/picoclock#2] [Q&A] [answered]
Hi all,

found this project and want to try on my rp2040 screen board.

cmake .. # passed flawlessly
make # fails with ld errors.

Could you help me to identify if it is error in my setup, or smth? Maybe I am missing some lib/devel package?

error looks like:
<pre>[ 95%] Building C object CMakeFiles/main.dir/home/ruslanas/git/pico-sdk/src/rp2_common/pico_unique_id/unique_id.c.obj
[ 96%] Building C object CMakeFiles/main.dir/home/ruslanas/git/pico-sdk/src/rp2_common/hardware_i2c/i2c.c.obj
[ 96%] Building C object CMakeFiles/main.dir/home/ruslanas/git/pico-sdk/src/rp2_common/hardware_spi/spi.c.obj
[ 97%] Building C object CMakeFiles/main.dir/home/ruslanas/git/pico-sdk/src/rp2_common/hardware_rtc/rtc.c.obj
[ 97%] Building C object CMakeFiles/main.dir/home/ruslanas/git/pico-sdk/src/rp2_common/hardware_adc/adc.c.obj
[ 98%] Building C object CMakeFiles/main.dir/home/ruslanas/git/pico-sdk/src/rp2_common/pico_bootsel_via_double_reset/pico_bootsel_via_double_reset.c.obj
[100%] Linking CXX executable main.elf
/usr/lib/gcc/arm-none-eabi/12.2.0/../../../../arm-none-eabi/bin/ld: CMakeFiles/main.dir/lib/draw.c.obj:(.bss.drawlines+0x0): multiple definition of `drawlines'; CMakeFiles/main.dir/main.c.obj:(.bss.drawlines+0x0): first defined here
/usr/lib/gcc/arm-none-eabi/12.2.0/../../../../arm-none-eabi/bin/ld: lib/lcd/liblcd.a(lcd.c.obj):(.data.drawlines+0x0): multiple definition of `drawlines'; CMakeFiles/main.dir/main.c.obj:(.bss.drawlines+0x0): first defined here
/usr/lib/gcc/arm-none-eabi/12.2.0/../../../../arm-none-eabi/bin/ld: warning: main.elf has a LOAD segment with RWX permissions
/usr/lib/gcc/arm-none-eabi/12.2.0/../../../../arm-none-eabi/bin/ld: warning: main.elf has a LOAD segment with RWX permissions
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/main.dir/build.make:1289: main.elf] Error 1
make[1]: *** [CMakeFiles/Makefile2:1423: CMakeFiles/main.dir/all] Error 2
make: *** [Makefile:91: all] Error 2
[ruslanas@ruslanas build]$
</pre>
Also checked for CMakeLists.txt file, and in the first line it is importing `pico_SDK_import.cmake`, which is located in project dir, BUT, from make output, I see it is using my copy of SDK, which looks to be newer and have `pico_sdk_init.cmake` for first include.

Did anyone faced similar issue?

Thanks in advance.

**Comment by dawigit** [comment, marked as answer]
Yes, Vlad had had it too. The linking error shows up when building on a Ubuntu system, and until now i built on a Pi4.
So when building on my Ubuntu system, I was able to reproduce the error and now I removed that obsolete vars/code.
Now building (and linking!) works. Commits already pushed. And as you proposed I created 'CHANGELOG.md'.
